src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java

```diff
@@ -65,17 +65,14 @@ public String id() {
         return "api-key";
     }
 
-    /**
-     * {@inheritDoc}
-     * The {@link ApiKeyPolicy} is assignable if an api key is passed in the request headers.
-     */
     @Override
-    public Single<Boolean> support(RequestExecutionContext ctx) {
-        final Optional<String> optApiKey = extractApiKey(ctx);
-
-        optApiKey.ifPresent(apiKey -> ctx.setInternalAttribute(ATTR_INTERNAL_API_KEY, apiKey));
-
-        return Single.just(optApiKey.isPresent());
+    public Maybe<SecurityToken> extractSecurityToken(RequestExecutionContext ctx) {
+        final Optional<String> apiKey = extractApiKey(ctx);
+        if (apiKey.isPresent()) {
+            ctx.setInternalAttribute(ATTR_INTERNAL_API_KEY, apiKey.get());
+            return Maybe.just(SecurityToken.forApiKey(apiKey.get()));
+        }
+        return Maybe.empty();
```

Comment:
I'm not sure but yu can maybe do something like:
```suggestion
        return extractApiKey(ctx)
        .map(apiKey -> {
            ctx.setInternalAttribute(ATTR_INTERNAL_API_KEY, apiKey);
            return Maybe.just(SecurityToken.forApiKey(apiKey));
        })
        .orElse(Maybe.empty());
```

Reply:
I think you're right.
Is that just for style ? (I find my way more readable, but just a personal opinion).